packages/astro/src/core/session.ts

```diff
@@ -455,6 +449,13 @@ export class AstroSession<TDriver extends SessionDriverName = any> {
 			this.#storage = createStorage({
 				driver: driver(this.#config.options),
 			});
+			if (this.#config.driver === 'fs-lite') {
+				// seems to be a race condition in unstorage when the session directory doesn't exist
+				// this ensures that the directory is created before we try to use it for real sessions
+				const dummyId = crypto.randomUUID();
+				await this.#storage.setItem(dummyId, 'init');
+				await this.#storage.removeItem(dummyId);
+			}
```

Comment:
OK, I think I understand this now. SAo is the problem that if you try to read a session when the ddirectory hasn't been created then it has an error? I'm not happy with this workaround because we're otherwise not using the storage unless the user actually reads or writes a session. This is soemthign that should be fixed upstream: unstorage should just return undefined for gets if the directory doesn't exist, the same as if the file doesn't exist. 

Have you encountered this issue outside of tests? If not, I think the best bet is to add a workaround inside tests, and report the issue on the unstorage repo.

Reply:
Glad you pushed back on this one - found the underlying bug. `PERSIST_SYMBOL` snuck through TypeScript's type inference and it didn't yell about it not being awaited.